test cases/common/43 has function/meson.build

```diff
@@ -71,10 +73,10 @@ foreach cc : compilers
            args : unit_test_args),
            'built-in alloca must be found with #include')
     if not cc.compiles(defines_has_builtin, args : unit_test_args)
-      assert(not cc.has_function('alloca',
+      assert(cc.has_function('alloca',
              prefix : '#include <alloca.h>\n#undef alloca',
              args : unit_test_args),
-             'built-in alloca must not be found with #include and #undef')
+             'built-in alloca must be found with #include and #undef')
```

Comment:
How does this work? It seems like it should not?

Reply:
The test was added to ensure that the `__builtin_foo` codepath is not hit when there is a `#include` in prefix because there was a `#error` in the other path. That other path is gone now, and we have a fallback check for `foo` there now.

`alloca` may not be a `#define` now, but it is still available as a symbol. See: `alloca.h`.